pkg/groupsync/manytomanysyncer.go

```diff
@@ -213,7 +213,10 @@ func (f *ManyToManySyncer) targetUsers(ctx context.Context, sourceUsers []*User)
 			merr = fmt.Errorf("error mapping source user id %s to target user id: %w", sourceUser.ID, err)
 			continue
 		}
-		targetUsers = append(targetUsers, &User{ID: targetUserID})
```

Comment:
What is the situation where we'd get back an empty string (but not an error)? That seems like a bug, no?

Reply:
If the target user mapping isn't found, we want to skip them and keep processing the rest of the users.

This was a bit of hacky way to allow the mapper to signal we are in this scenario. 

Probably a more proper way would be define a custom error, something like `targetUserNotFound = errors.new("target user not found")` and then the userMapper can return that error for that situation, and we can check for that error and then skip.

Reply:
Oh and if your wondering why this is needed at all since this function continues processing the users after an error, the entire sync is aborted if an error is returned from this function:
https://github.com/abcxyz/team-link/blob/main/pkg/groupsync/manytomanysyncer.go#L134

So we need a way to skip the user without returning an error.

Reply:
I would prefer if we returned an error that we could check using `errors.Is`. Returning the empty string is an API that will have unintended consequences in the future and be difficult to debug.

Reply:
Fair enough. Always keeping me honest @sethvargo :smile: